tests/Integration/RestTest.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Parsely REST API tests.
+ *
+ * @package Parsely\Tests
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\Tests\Integration;
+
+use Parsely\Parsely;
+use Parsely\Rest;
+
+
+/**
+ * Parsely REST API tests.
+ */
+final class RestTest extends TestCase {
+	/**
+	 * Internal variable
+	 *
+	 * @var Rest $rest Holds the Rest object
+	 */
+	private static $rest;
+
+	/**
+	 * The setUp run before each test
+	 */
+	public function set_up(): void {
+		parent::set_up();
+
+		self::$rest = new Rest( new Parsely() );
+	}
+
+	/**
+	 * Test whether the logic has been enqueued in the `rest_api_init` hook.
+	 *
+	 * @covers \Parsely\Rest\run;
+	 */
+	public function test_register_enqueued_rest_init() {
+		self::$rest->run();
+		$this->assertSame(
+			10,
+			has_action( 'rest_api_init', array( self::$rest, 'register_parsely_meta' ) )
+		);
+	}
+
+	/**
+	 * Test whether the logic has been enqueued in the `rest_api_init` hook with a filter that disables it.
+	 *
+	 * @covers \Parsely\Rest\run;
+	 */
+	public function test_register_enqueued_rest_init_filter() {
+		add_filter( 'wp_parsely_enable_rest_api_support', '__return_false' );
+		self::$rest->run();
+		$this->assertSame(
+			false,
+			has_action( 'rest_api_init', array( self::$rest, 'register_parsely_meta' ) )
+		);
+	}
+
+	/**
+	 * Test that the REST fields are registered to WordPress REST API.
+	 *
+	 * @covers \Parsely\Rest\register_parsely_meta
+	 */
+	public function test_register_parsely_meta_registers_fields() {
+		global $wp_rest_additional_fields;
+
+		self::$rest->register_parsely_meta();
+
+		$expected_fields = array( 'get_callback', 'update_callback', 'schema' );
```

Comment:
Instead of checking that the array_keys of `$wp_rest_additional_fields['...']['parsely']` are these three values (which, they [always will be](https://github.com/WordPress/WordPress/blob/47c4ff7102291136473d1ae52302dd6ebfad3345/wp-includes/rest-api.php#L137) - just with different callback values, with a default of `null`), why not check that the parsely property exists for that post type in the global, and that the callback is not the default `null`?

```php
isset( $wp_rest_additional_fields['post']['parsely'] ) && ! is_null( $wp_rest_additional_fields['post']['parsely']['get_callback']
```

There's also an opportunity to check that `['get_callback']` is populated correctly - an array, with a `version` key being a string, and `meta` key with an array which is at least going to contain some expected fields for particular post types (post vs non-post).